Documentation/gettingstarted/dns.rst

```diff
@@ -0,0 +1,203 @@
+
+.. parsed-literal::
+
+    $ kubectl create -f \ |SCM_WEB|\/examples/minikube/dns-sw-app.yaml
+    $ kubectl get po,svc
+    NAME                             READY   STATUS    RESTARTS   AGE
+    pod/deathstar-5fc7c7795d-7ljr8   1/1     Running   0          14s
+    pod/deathstar-5fc7c7795d-lvw69   1/1     Running   0          14s
+    pod/mediabot                     1/1     Running   0          14s
+    pod/spaceship                    1/1     Running   0          14s
+
+    NAME                 TYPE        CLUSTER-IP       EXTERNAL-IP   PORT(S)   AGE
+    service/deathstar    ClusterIP   10.110.127.146   <none>        80/TCP    14s
+    service/kubernetes   ClusterIP   10.96.0.1        <none>        443/TCP   16m
+
+Step 4: Check Current Access
+============================
+
+Since we have not enforced any policies, both ``mediabot`` and ``spaceship`` have access to twitter API service as well as internal ``deathstar`` service. You can confirm this by running the following commands.
+
+.. parsed-literal::
+
+    $ kubectl exec -it mediabot -- curl -sL https://api.twitter.com
+    ...
+
+    $ kubectl exec -it spaceship -- curl -sL https://api.twitter.com
+    ...
+
+    $ kubectl exec -it spaceship -- curl -sL deathstar.default.svc.cluster.local/v1
+    {
+	"name": "Death Star",
+        ...
+        ...
+    }
+
+    $ kubectl exec -it mediabot -- curl -sL deathstar.default.svc.cluster.local/v1
+    {   
+        ...
+        ...
+    }
+
+Step 5: Apply DNS Egress Policy
+===============================
+
+We will enforce a Cilium network policy which allows:
+
+* Only the ``mediabot`` pods have access to ``api.twitter.com``. 
+* Both the pods have access to the internal ``deathstar`` service.
+
+The following Cilium network policy achieves above access controls. 
+
+.. literalinclude:: ../../examples/minikube/dns-policy.yaml
+
+Let's take a closer look at the policy: 
+
+* The first egress section uses ``toFQDNs: matchName`` specification to allow egress to ``api.twitter.com``. The destination DNS should match exactly the name specified in the rule. The ``endpointSelector`` allows only pods with labels ``class: mediabot`` to have the egress access.
+* The second egress section allows all pods in the ``default`` namespace to access ``kube-dns`` service. Important to note the part ``rules: dns`` that specifies Cilium to inspect and allow DNS lookups matching specified patterns. In this case, inspect and allow all DNS queries. 
+* The third egress section allows all pods in the ``default`` namespace to access the internal ``deathstar`` service. Note that for controlling access to Kubernetes services we use the labels to identify the destination service rather than use DNS. 
+
+Let's apply the policy:
+
+.. parsed-literal::
+
+  $ kubectl create -f \ |SCM_WEB|\/examples/minikube/dns-policy.yaml
```

Comment:
This also needs a more descriptive filename

Reply:
changed to `dns-matchname.yaml`